bin/github-linguist

```diff
@@ -32,19 +33,20 @@ def github_linguist(args)
       puts opts
       exit
     end
+    opts.on("-s", "--stdin", "Read file contents from stdin") { read_from_stdin = true }
```

Comment:
Conventionally, programs which accept a filename as an argument are made to read from STDIN by passing `-` as a parameter. Linguist should support this idiom as well.

Reply:
I don't see how Linguist can adopt that idiom without making it even more confusing as Linguist still needs the filename for language detection. If we replace the filename with `-` then we'd need to create an additional parameter such as `--filename`

Reply:
@mschuldt See my latest comment.